environment.lua

```diff
@@ -93,9 +93,10 @@ local function log_binding(name, type, value)
 end
 
 ---@overload fun(binding: binding) : boolean, string
+---@param span Span
 ---@param binding binding
 ---@return boolean, Environment
-function environment:bind_local(binding)
+function environment:bind_local(span, binding)
```

Comment:
For now, I'd prefer leading each member of `terms.binding` with a `cause_span` (spanning the cause of the binding), at least until we get `terms.spanned_binding` containing a `terms.binding`.